### DISCUSSION TITLE
problems installing 3.0.1 with docker

### DISCUSSION THREAD

**Discussion by kesself** [ucfcdl/UDOIT#842] [Q&A] [answered]
I am trying to install 3.0.1 with docker

When I run 
[root@lti1-lms-dev UDOIT-3.0.1]# docker-compose up --build
[+] Building 3.7s (16/20)                                                                                                                                                              
 => [internal] load build definition from Dockerfile                                                                                                                              0.0s
 => => transferring dockerfile: 85B                                                                                                                                               0.0s
 => [internal] load .dockerignore                                                                                                                                                 0.0s
 => => transferring context: 2B                                                                                                                                                   0.0s
 => [internal] load metadata for docker.io/library/php:7.4-fpm                                                                                                                    0.8s
 => FROM docker.io/library/composer:latest                                                                                                                                        0.5s
 => => resolve docker.io/library/composer:latest                                                                                                                                  0.5s
 => [internal] load build context                                                                                                                                                 0.0s
 => => transferring context: 25.47kB                                                                                                                                              0.0s
 => [stage-0  1/15] FROM docker.io/library/php:7.4-fpm@sha256:94df786c69e98af514a5399c788266e9c79b2cc863975ceaf21ac46b11850722                                                    0.0s
 => CACHED [stage-0  2/15] RUN apt-get update && apt-get install -y         git         libfreetype6-dev         libjpeg62-turbo-dev         libpng-dev         libpq-dev         0.0s
 => CACHED [stage-0  3/15] RUN if [ "$ENVIORNMENT_TYPE" != "local" ] ;then          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"         &&  0.0s
 => CACHED [stage-0  4/15] RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -     && apt-get update && apt-get install -y nodejs                                         0.0s
 => CACHED [stage-0  5/15] RUN npm install --global yarn                                                                                                                          0.0s
 => CACHED [stage-0  6/15] RUN echo "ServerName localhost" >> /etc/apache2/apache2.conf                                                                                           0.0s
 => CACHED [stage-0  7/15] RUN apachectl start                                                                                                                                    0.0s
 => CACHED [stage-0  8/15] RUN useradd -ms /bin/bash ssm-user                                                                                                                     0.0s
 => CACHED [stage-0  9/15] RUN mkdir -p /var/www/html     && chown ssm-user:www-data /var/www/html                                                                                0.0s
 => CACHED [stage-0 10/15] COPY --from=composer:latest /usr/bin/composer /usr/local/bin/composer                                                                                  0.0s
 => ERROR [stage-0 11/15] RUN wget https://get.symfony.com/cli/installer -O - | bash &&     mv /root/.symfony/bin/symfony /usr/local/bin/symfony                                  2.2s
------
 > [stage-0 11/15] RUN wget https://get.symfony.com/cli/installer -O - | bash &&     mv /root/.symfony/bin/symfony /usr/local/bin/symfony:
#0 0.572 --2022-08-24 17:42:38--  https://get.symfony.com/cli/installer
#0 0.579 Resolving get.symfony.com (get.symfony.com)... 54.192.76.79, 54.192.76.87, 54.192.76.23, ...
#0 0.699 Connecting to get.symfony.com (get.symfony.com)|54.192.76.79|:443... connected.
#0 0.849 HTTP request sent, awaiting response... 200 OK
#0 0.887 Length: 6100 (6.0K) [binary/octet-stream]
#0 0.887 Saving to: 'STDOUT'
#0 0.888 
#0 0.888      0K .....                                                 100%  233M=0s
#0 0.888 
#0 0.888 2022-08-24 17:42:38 (233 MB/s) - written to stdout [6100/6100]
#0 0.888 
#0 0.895 Symfony CLI installer
#0 0.895 
#0 0.895 Environment check
#0 0.895   [*] cURL is installed
#0 0.896   [*] Tar is installed
#0 0.896   [*] Git is installed
#0 0.901   [*] Your architecture (amd64) is supported
#0 0.901 
#0 0.901 Download
#0 0.904   Downloading https://github.com/symfony-cli/symfony-cli/releases/latest/download/symfony-cli_linux_amd64.tar.gz...
#0 0.911   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#0 0.912                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 5105k  100 5105k    0     0  5241k      0 --:--:-- --:--:-- --:--:--  246M
#0 1.888   Uncompress binary...
#0 1.998   Installing the binary into your home directory...
#0 2.002   The binary was saved to: /root/.symfony5/bin/symfony
#0 2.002 
#0 2.003 The Symfony CLI was installed successfully!
#0 2.003 
#0 2.003 Use it as a local file:
#0 2.003   /root/.symfony5/bin/symfony
#0 2.004 
#0 2.004 Or add the following line to your shell configuration file:
#0 2.004   export PATH="$HOME/.symfony5/bin:$PATH"
#0 2.004 
#0 2.004 Or install it globally on your system:
#0 2.004   mv /root/.symfony5/bin/symfony /usr/local/bin/symfony
#0 2.004 
#0 2.004 Then start a new shell and run 'symfony'
#0 2.006 mv: cannot stat '/root/.symfony/bin/symfony': No such file or directory
------
failed to solve: executor failed running [/bin/sh -c wget https://get.symfony.com/cli/installer -O - | bash &&     mv /root/.symfony/bin/symfony /usr/local/bin/symfony]: exit code: 1

I have tried running with sudo and I get the same errors.....this is on Oracle Linux 8

Any help appreciated

**Comment by bagofarms** [comment, marked as answer]
I recommend pulling the latest code from `main`, which is 3.3.0, and following the new directions for Docker in INSTALL.md.  The old Docker setup was pretty error-prone.

**Reply by bagofarms** [reply]
The structure of the project has changed. Pull the latest `stable/*` branch. Right now, that's `stable/3.3.x`.
